File: stack/python/stack.py

class Stack:
    def __init__(self):
        self.items = []

    # push - O(1)
    def push(self, item):
        self.items.append(item)

    # pop - O(1)
    def pop(self):
        return self.items.pop()

    # size
    def size(self):
        return len(self.items)

    # is empty
    def is_empty(self):
        return self.items == []

    # peek
    def peek(self):
        return self.items[len(self.items)-1]

File: queues_stacks/stack/stack.js

// We almost always can just use a normal JS array to implement this
// without writing a class like the following.

class Stack {

  constructor() {
    this.list = [];
    this.length = 0;
  }

  push(value) {
    this.length++;
    this.list.push(value);
  }

  pop() {
    if (this.length === 0) return;

    // Pop the last item off the end of the list and return the value.
    this.length--;
    return this.list.pop();
  }

  peek() {
    return this.list[this.length - 1];
  }
}
